src/analysis/buildTypeWeights.ts

```diff
@@ -193,18 +220,95 @@ function parseTypes(schema: GraphQLSchema, typeWeights: TypeWeightSet): TypeWeig
                     weight: typeWeights.scalar,
                 };
             } else if (isUnionType(currentType)) {
-                // FIXME: will need information on fields inorder calculate comlpextiy
-                result[typeName] = {
-                    fields: {},
-                    weight: typeWeights.object,
-                };
+                unions.push(currentType);
             } else {
+                // FIXME: Scalar types are listed here throw new Error(`ERROR: buildTypeWeight: Unsupported type: ${currentType}`);
                 // ? what else can get through here
                 // ? inputTypes?
             }
         }
     });
 
+    unions.forEach((unionType: GraphQLUnionType) => {
+        /** Start with the fields for the first object. Store fieldname, type, weight and resolve to for later use
+         *  reduce by selecting fields common to each type
+         *  compare both fieldname and output type accounting for lists and non-nulls
+         *      for object
+         *          compare name of output type
+         *      for lists
+         *          compare ofType and ofType name if not onother list/non-null
+         *      for non-nulls
+         *          compare oftype and ofTypeName (if not another non-null)
+         *  */
```

Comment:
Just sayin', this comment is kind of cryptic. I can follow what's going on if I look at the code for long enough but this comment didn't help. It didn't hurt, but it didn't help. You could probably do a better job of explaining your strategy here for future engineers.

Reply:
Very nice. This much more clearly express what I thought was going on.